pkg/builder/builder.go

```diff
@@ -85,32 +76,35 @@ func (b *Builder) PrepareBuild(bldr *Builder, mfst *manifest.Manifest, appDir st
 		Credentials: ctx.Manifest.Credentials,
 	}
 
-	for _, c := range ctx.Components {
-		if err := c.PrepareBuild(ctx); err != nil {
+	for _, imb := range imageBuilders {
+		registry := ctx.Manifest.InvocationImages[imb.Name()].Configuration["registry"]
+		if err := imb.PrepareBuild(ctx.AppDir, registry, ctx.Manifest.Name); err != nil {
 			return nil, nil, err
 		}
 
-		if c.Name() == "cnab" {
-			ii := bundle.InvocationImage{}
-			ii.Image = c.URI()
-			ii.ImageType = c.Type()
-			bf.InvocationImages = []bundle.InvocationImage{ii}
-			//bf.Version = strings.Split(c.URI(), ":")[1]
-			baseVersion := mfst.Version
-			if baseVersion == "" {
-				baseVersion = "0.1.0"
-			}
-			newver, err := b.version(baseVersion, strings.Split(c.URI(), ":")[1])
-			if err != nil {
-				return nil, nil, err
-			}
-			bf.Version = newver
-		} else {
-			bundleImage := bundle.Image{Description: c.Name()}
-			bundleImage.Image = c.URI()
-			bundleImage.ImageType = c.Type()
-			bf.Images = append(bf.Images, bundleImage)
+		ii := bundle.InvocationImage{}
+		ii.Image = imb.URI()
+		ii.ImageType = imb.Type()
+		bf.InvocationImages = append(bf.InvocationImages, ii)
+
+		baseVersion := mfst.Version
+		if baseVersion == "" {
+			baseVersion = "0.1.0"
+		}
+		newver, err := b.version(baseVersion, strings.Split(imb.URI(), ":")[1])
+		if err != nil {
+			return nil, nil, err
 		}
+		bf.Version = newver
+	}
+
+	for _, image := range mfst.Images {
+		//TODO don't need imagebuilders for images anymore so not sure what this looks like
```

Comment:
Let me see if I can recall the context for this:

- we decided that the builder is only responsible for building invocation images
- as discussed in #564, we also want to add other images to the bundle (for example images we don't build / are already built)
- we could use this to iterate on the Duffle manifest referenced images and add them to the bundle?